src/Konami.tsx

```diff
@@ -0,0 +1,167 @@
+import React from 'react';
+
+export interface KonamiProps {
+  className?: string;
+  code?: number[];
+  disabled?: boolean;
+  resetDelay?: number;
+  action?: () => void;
+  timeout?: number;
+  onTimeout?: () => void;
+}
+
+export interface KonamiStates {
+  done: boolean;
+  input: number[];
+}
+
+class Timer {
+  t: number;
+  timerObj: number | null;
+  fn: TimerHandler;
+
+  constructor(fn: () => void, t: number) {
+    this.t = t;
+    this.fn = fn;
+
+    this.timerObj = setInterval(this.fn, this.t);
+  }
+
+  stop() {
+    if (this.timerObj) {
+      clearInterval(this.timerObj);
+      this.timerObj = null;
+    }
+    return this;
+  }
+
+  start() {
+    if (!this.timerObj) {
+      this.stop();
+      this.timerObj = setInterval(this.fn, this.t);
+    }
+    return this;
+  }
+
+  reset(newT: number) {
+    this.t = newT;
+    return this.stop().start();
+  }
+}
+
+class Konami extends React.Component<KonamiProps, KonamiStates> {
+  private timeoutFunc: ReturnType<typeof setTimeout> | null | undefined;
+  private _timer: any;
+
+  static defaultProps = {
+    action: null,
+    className: '',
+    code: [38, 38, 40, 40, 37, 39, 37, 39, 66, 65],
+    disabled: false,
+    onTimeout: null,
+    resetDelay: 1000,
+    timeout: null,
+  };
+
+  constructor(props: KonamiProps) {
+    super(props);
+
+    this.state = {
+      done: false,
+      input: [],
+    };
+
+    this.timeoutFunc = null;
```

Comment:
Same for `timeoutFunc`, I could've named it `timeoutID`.

Reply:
done :)